src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let total = 0;
+
+  function adder(...args) {
+    if (args.length === 0) {
+      const result = total;
+
+      total = 0;
+
+      return result;
+    } else {
+      total += args.reduce((sum, num) => sum + num, 0);
```

Comment:
Since the function is only supposed to take one argument at a time, the use of `reduce` is not appropriate. You should add the single argument directly to the `total`.